File: src/sys/java/fan/sys/UnresolvedErr.java
//
// Copyright (c) 2006, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   4 Jan 06  Brian Frank  Creation
//
package fan.sys;

/**
 * UnresolvedErr
 */
public class UnresolvedErr
  extends Err
{

//////////////////////////////////////////////////////////////////////////
// Java Convenience
//////////////////////////////////////////////////////////////////////////

  public static UnresolvedErr make(String msg, Throwable cause)  { return make(msg, Err.make(cause)); }
  public static UnresolvedErr make(Uri uri)  { return make(uri.str); }

//////////////////////////////////////////////////////////////////////////
// Fantom Constructors
//////////////////////////////////////////////////////////////////////////

  public static UnresolvedErr make() { return make("", (Err)null); }
  public static UnresolvedErr make(String msg) { return make(msg, (Err)null); }
  public static UnresolvedErr make(String msg, Err cause)
  {
    UnresolvedErr err = new UnresolvedErr();
    make$(err, msg, cause);
    return err;
  }

  public static void make$(UnresolvedErr self) { make$(self, null);  }
  public static void make$(UnresolvedErr self, String msg) { make$(self, msg, null); }
  public static void make$(UnresolvedErr self, String msg, Err cause) { Err.make$(self, msg, cause); }

//////////////////////////////////////////////////////////////////////////
// Java Constructors
//////////////////////////////////////////////////////////////////////////

  public UnresolvedErr(Err.Val val) { super(val); }
  public UnresolvedErr() { super(new UnresolvedErr.Val()); }

//////////////////////////////////////////////////////////////////////////
// Identity
//////////////////////////////////////////////////////////////////////////

  public Type typeof() { return Sys.UnresolvedErrType; }

//////////////////////////////////////////////////////////////////////////
// Val - Java Exception Type
//////////////////////////////////////////////////////////////////////////

  public static class Val extends Err.Val {}

}

File: src/sys/dotnet/fan/sys/UnresolvedErr.cs
//
// Copyright (c) 2007, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   28 Nov 07  Andy Frank  Creation
//

using System;
using System.IO;

namespace Fan.Sys
{
  /// <summary>
  /// UnresolvedErr
  /// </summary>
  public class UnresolvedErr : Err
  {

  //////////////////////////////////////////////////////////////////////////
  // C# Convenience
  //////////////////////////////////////////////////////////////////////////

    public new static UnresolvedErr make(string msg, Exception cause)  { return make(msg, Err.make(cause)); }
    public static UnresolvedErr make(Uri uri) { return make(uri.m_str); }

  //////////////////////////////////////////////////////////////////////////
  // Fantom Constructors
  //////////////////////////////////////////////////////////////////////////

    public new static UnresolvedErr make() { return make("", (Err)null); }
    public new static UnresolvedErr make(string msg) { return make(msg, (Err)null); }
    public new static UnresolvedErr make(string msg, Err cause)
    {
      UnresolvedErr err = new UnresolvedErr();
      make_(err, msg, cause);
      return err;
    }

    public static void make_(UnresolvedErr self) { make_(self, null);  }
    public static void make_(UnresolvedErr self, string msg) { make_(self, msg, null); }
    public static void make_(UnresolvedErr self, string msg, Err cause) { Err.make_(self, msg, cause); }

  //////////////////////////////////////////////////////////////////////////
  // C# Constructors
  //////////////////////////////////////////////////////////////////////////

    public UnresolvedErr(Err.Val val) : base(val) {}
    public UnresolvedErr() : base(new UnresolvedErr.Val()) {}
    public UnresolvedErr(Exception actual) : base(new UnresolvedErr.Val(), actual) {}

  //////////////////////////////////////////////////////////////////////////
  // Identity
  //////////////////////////////////////////////////////////////////////////

    public override Type @typeof() { return Sys.UnresolvedErrType; }

  //////////////////////////////////////////////////////////////////////////
  // Val - C# Exception Type
  //////////////////////////////////////////////////////////////////////////

    public new class Val : Err.Val {}

  }
}